src/daemon/daemon.cpp

```diff
@@ -2405,13 +2698,13 @@ void mp::Daemon::persist_instances()
     }
     QDir data_dir{
         mp::utils::backend_directory_path(config->data_directory, config->factory->get_backend_directory_name())};
-    mp::write_json(instance_records_json, data_dir.filePath(instance_db_name));
+    MP_JSONUTILS.write_json(instance_records_json, data_dir.filePath(instance_db_name));
 }
 
 void mp::Daemon::release_resources(const std::string& instance)
 {
-    config->factory->remove_resources_for(instance);
     config->vault->remove(instance);
+    config->factory->remove_resources_for(instance);
```

Comment:
Has this been a latent bug where this should be called after `remove()` or did this become necessary with snapshots in mind?

Reply:
This was necessary for some reason, I forget exactly why. @sharder996, do you remember why this was necessary? 

Oh I think it may have to do with the fact that instances now know about their directory and could try to persist snapshots after it was removed...

Reply:
Initially, it was the other way around with `config->vault->remove()` needing to be called last as it removed the instance directory and its contents.

Now that that dependency on order of operations has been moved [here](https://github.com/canonical/multipass/blob/eebe75f2f92c27b2ccf88a91b3d2b1aa3f448fb0/src/platform/backends/shared/base_virtual_machine_factory.h#L98:L103), the above lines have been reordered to make more conceptual sense. As the vault does not, and should not, depend on a specific instance, the order of these function calls no longer matters.

Reply:
Ah, ok, thanks for the explanation @sharder996.